planner/core/logical_plan_builder.go

```diff
@@ -2841,7 +2866,9 @@ func (b *PlanBuilder) buildDelete(delete *ast.DeleteStmt) (Plan, error) {
 		}
 	}
 
-	return del, nil
+	err = del.ResolveIndices()
```

Comment:
Can we not do this?

Reply:
No, we need to resolve the index for the handle column the delete executor holds.